xbmc/guilib/GUIAction.h

```diff
@@ -16,26 +16,74 @@ class CGUIControl;
 class CGUIListItem; typedef std::shared_ptr<CGUIListItem> CGUIListItemPtr;
 
 /**
- * Class containing vector of condition->(action/navigation route) pairs and handling its execution.
+ * Class containing vector of condition->(action/navigation route) and handling its execution.
  */
 class CGUIAction
 {
 public:
+  /**
+   * Structure containing the the executable action definition
+   */
+  struct ExecutableAction
```

Comment:
Should be a `class` - you fully encapsulated members, which is good.

Reply:
What? Not really, see https://stackoverflow.com/questions/92859/what-are-the-differences-between-struct-and-class-in-c

Reply:
What? Not really, see https://stackoverflow.com/questions/92859/what-are-the-differences-between-struct-and-class-in-c